code/modules/mob/living/living_movement.dm

```diff
@@ -147,7 +147,15 @@ default behaviour is:
 				now_pushing = 0
 				return
 
-		step(AM, t)
+		var/turf/T = AM.loc
+		var/turf/T2 = get_step(AM,t)
+		if(!T2) // Map edge
+			now_pushing = 0
+			return
+		var/move_time = movement_delay(loc, t)
```

Comment:
movement_delay is a tick-time between movements where higher is slower. `* sqrt(2) / 2` comes out at a fraction of 1, making diagonal movements faster compared to cardinal. Removing the / 2 is then correct for euclidean distance - but it is also much preferable to implement this with a constant:
```
var/global/const/SQRT_TWO = 1.41421356237
...
var/move_time = movement_delay(loc, t) * SQRT_TWO
```


As to whether euclidean distance is a reasonable multiplier, I disagree. People *not* moving diagonally are then actively disadvantaged, because they take the full time of two turf changes to move. Most reasonable is simply to consider a diagonal movement the same as its component x/y, and multiply move time by two:
```
var/move_time = movement_delay(loc, t) * 2
```